nfc/libs/NfcCoreLib/lib/Nci/phNciNfc.c

```diff
@@ -1715,10 +1721,11 @@ NFCSTATUS phNciNfc_GetNfccFeatures(void *pNciCtx,
         /* Store the Manufacturer ID */
         pNfccFeatures->ManufacturerId = pNciContext->InitRspParams.ManufacturerId;
         /* Store the Manufacturer specific info */
-        pNfccFeatures->ManufactureInfo.Byte0 = pNciContext->InitRspParams.ManufacturerInfo.Byte0;
-        pNfccFeatures->ManufactureInfo.Byte1 = pNciContext->InitRspParams.ManufacturerInfo.Byte1;
-        pNfccFeatures->ManufactureInfo.Byte2 = pNciContext->InitRspParams.ManufacturerInfo.Byte2;
-        pNfccFeatures->ManufactureInfo.Byte3 = pNciContext->InitRspParams.ManufacturerInfo.Byte3;
+        pNfccFeatures->ManufactureInfo.ManufacturerInfoLength =
+            pNciContext->InitRspParams.ManufacturerInfo.ManufacturerInfoLength;
+        pNfccFeatures->ManufactureInfo.ManufacturerInfoBuffer =
+            pNciContext->InitRspParams.ManufacturerInfo.ManufacturerInfoBuffer;
+
```

Comment:
I think i don't need to copy the buffer. Goal of those lines is to save the buffer address allocated in phNciNfc_ProcessInitRspNci1x or in phNciNfc_ResetNtfCb.
Manufacturer Information data are saved across different structures and files (phNciNfc.h, phNciNfc_Core.h and phLibNfc.h). The idea is to allocate and copy only once Manufacturer Information data and save buffer address in the relevant structures.

In the current code and after thinking about your feedback twice :-)... I believe, there is potentially a memory leak on that buffer in the following cases:
- More than 1 Core Reset Ntf raised in the same "session". A new buffer will be created loosing the previous one.
- A potential buggy firmware raising a Core Reset Ntf following the Nci 2.0 specification and a Core Init Rsp following the Nci 1.x specification...

I will work on adding a check before the buffer allocation where relevant (phNciNfc_ProcessInitRspNci1x and phNciNfc_ResetNtfCb).